core/src/main/scala/com.snowplowanalytics.stream/loader/Config.scala

```diff
@@ -161,92 +130,107 @@ object Config {
-            deriveReader[ESCluster]
-        }
-
-        case class ESChunk(byteLimit: Long, recordLimit: Long)
-
-        object ESChunk {
-          implicit val chunkConfigReader: ConfigReader[ESChunk] = deriveReader[ESChunk]
-        }
+        final case class ESChunk(byteLimit: Long, recordLimit: Long)
       }
-
-      implicit val goodSinkConfigReader: ConfigReader[GoodSink] = deriveReader[GoodSink]
     }
 
     sealed trait BadSink extends Product with Serializable
     object BadSink {
-      case class None() extends BadSink
-      object None {
-        implicit val badSinkNoneConfigReader: ConfigReader[None] = deriveReader[None]
-      }
+      final case object None extends BadSink
 
-      case class Stderr() extends BadSink
-      object Stderr {
-        implicit val badSinkStderrConfigReader: ConfigReader[Stderr] = deriveReader[Stderr]
-      }
+      final case object Stderr extends BadSink
 
-      case class Nsq(
+      final case class Nsq(
         streamName: String,
         nsqdHost: String,
         nsqdPort: Int
       ) extends BadSink
-      object Nsq {
-        implicit val badSinkNsqConfigReader: ConfigReader[Nsq] = deriveReader[Nsq]
-      }
 
-      case class Kinesis(
+      final case class Kinesis(
         streamName: String,
         region: String,
         customEndpoint: Option[String]
       ) extends BadSink {
         val endpoint = getKinesisEndpoint(region, customEndpoint)
       }
-      object Kinesis {
-        implicit val badSinkKinesisConfigReader: ConfigReader[Kinesis] = deriveReader[Kinesis]
-      }
-
-      implicit val badSinkConfigReader: ConfigReader[BadSink] = deriveReader[BadSink]
     }
   }
 
   sealed trait Purpose extends EnumEntry with EnumEntry.Hyphencase
   object Purpose extends Enum[Purpose] {
-    case object Good      extends Purpose
-    case object Bad       extends Purpose
-    case object PlainJson extends Purpose
+    final case object Good      extends Purpose
+    final case object Bad       extends Purpose
+    final case object PlainJson extends Purpose
 
     val values = findValues
 
-    implicit val purposeReader: ConfigReader[Purpose] = deriveEnumerationReader[Purpose]
+    implicit val purposeConfigReader: ConfigReader[Purpose] = deriveEnumerationReader[Purpose]
   }
 
-  case class StreamLoaderConfig(
-    input: Source,
-    output: Sink,
-    purpose: Purpose,
-    monitoring: Option[Monitoring]
+  final case class Monitoring(
+    snowplow: Monitoring.SnowplowMonitoring,
+    metrics: Option[Monitoring.Metrics]
   )
+  object Monitoring {
+    final case class SnowplowMonitoring(
+      collector: String,
+      appId: String
+    )
 
-  object StreamLoaderConfig {
-    implicit val streamLoaderConfigReader: ConfigReader[StreamLoaderConfig] =
-      deriveReader[StreamLoaderConfig]
+    final case class Metrics(cloudWatch: Boolean)
   }
 
+  implicit val streamLoaderConfigReader: ConfigReader[StreamLoaderConfig] =
```

Comment:
I moved all the implicits to one place [similar to rdb loader](https://github.com/snowplow/snowplow-rdb-loader/blob/master/modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/config/Config.scala#L184). I think it is cleaner like this. Since these implicits are only used by config parser, I guess we wouldn't have problem about their discoverability.

Reply:
@istreeter May I ask are you okay with this change ?